Q12.py

```diff
@@ -0,0 +1,45 @@
+class BankAccount:
+    def __init__(self, acc_name, acc_number, balance, pin):
+        self.acc_name = acc_name
+        self.acc_number=acc_number
+        self.balance=balance
+        self.pin=pin
+    
+    def check_pin(self, pin):
+        #Checking the pin on bases of condition and then giving account information
+        if self.pin == pin:
+            return "Account Holder Name: {}\nAccount Number: {}\nTotal Balance: {}".format(self.acc_name, self.acc_number, self.balance)
```

Comment:
Why you are using same code to check pin again and again if you have created check_pin function? Please update it.